src/Bicep.Core/Registry/OciArtifactRegistry.cs

```diff
@@ -382,6 +398,29 @@ protected override void WriteArtifactContentToCache(OciArtifactReference referen
                 }
             }
 
+            if (result is OciProviderArtifactResult providerArtifact)
+            {
+                var layerName = RuntimeInformation.ProcessArchitecture switch {
+                    Architecture.X64 when RuntimeInformation.IsOSPlatform(OSPlatform.Linux) => BicepMediaTypes.BicepProviderArtifactLayerV1LinuxX64Binary,
+                    Architecture.Arm64 when RuntimeInformation.IsOSPlatform(OSPlatform.OSX) => BicepMediaTypes.BicepProviderArtifactLayerV1OsxArm64Binary,
+                    Architecture.X64 when RuntimeInformation.IsOSPlatform(OSPlatform.Windows) => BicepMediaTypes.BicepProviderArtifactLayerV1WinX64Binary,
+                    _ => throw new InvalidOperationException($"Unsupported architecture: {RuntimeInformation.ProcessArchitecture}"),
+                };
```

Comment:
The OCI artifact will need to contain some form of metadata to allow us to distinguish "this is a types-only package" from "this is a binary & types package, but your architecture isn't supported"